apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+     */
+    function depositToFinance(address token) public auth(OWNER_WRITE_ROLE) {
+        ERC20 tokenContract = ERC20(token);
+        uint256 value = tokenContract.balanceOf(this);
+        require(value >= 0);
+
+        // make an aprrovement for the same value to Finance
+        tokenContract.approve(address(finance), value);
+        // finally deposit those tokens to Finance
+        finance.deposit(tokenContract, value, "Adding Funds");
+    }
+
+    /**
+     * @dev Implement escape hatch mechanism. Avoids locked in contract forever
+     * @notice Implement escape hatch mechanism. Avoids locked in contract forever
+     */
+    function escapeHatch() public auth(OWNER_WRITE_ROLE) {
+        for (uint i = 0; i < globalTokensArray.length; i++) {
+            var token = globalTokensArray[i];
+            ERC20 tokenContract = ERC20(token);
+            uint256 tokenBalance = tokenContract.balanceOf(this);
+            // Avoid zero balances
+            if (tokenBalance == 0)
+                continue;
+            // send balance to owner
+            if (token != address(etherToken)) {
+                tokenContract.transfer(employer, tokenBalance);
+            } else {
+                etherToken.withdraw(employer, tokenBalance); // withdraw ether to receiver
+            }
+            LogFund(
+                employer,
+                token,
+                tokenBalance,
+                tokenContract.balanceOf(this),
+                ""
+            );
+        }
+        selfdestruct(employer); // send funds to organizer
+    }
+
+    /**
+     * @dev Implement escape hatch mechanism for critical case. Avoids locked in contract forever
+     * @notice Implement escape hatch mechanism for critical case.
+               Avoids locked in contract forever. If above function fails
+               (imagine, e.g., a bad token with a failing implementation of transfer),
+               this one would at least recover ETH (which btw should never remain in this contract).
+     */
+    function escapeHatch2() public auth(OWNER_WRITE_ROLE) {
+        selfdestruct(employer); // send funds to organizer
+    }
+
+    /**
+     * @dev To be able to receive ERC20 Token transfers, using approveAndCall
+     *      See, e.g: https://www.ethereum.org/token
+     * @notice To be able to receive ERC20 Token transfers, using approveAndCall
+     * @param from  Token sender address.
+     * @param value Amount of tokens.
+     * @param token Token to be received.
+     * @param data  Transaction metadata.
+     */
+    function receiveApproval(
+        address from,
+        uint256 value,
+        address token,
+        bytes data
+    )
+        public
+        returns(bool success)
+    {
+        ERC20 tokenContract = ERC20(token);
+
+        // first send tokens to this contract
+        tokenContract.transferFrom(from, this, value);
```

Comment:
add a require in this function to ensure the transfer went well

Reply:
Sure.